src/Symfony/Component/String/AbstractUnicodeString.php

```diff
@@ -142,8 +142,10 @@ public function ascii(array $rules = []): self
             } elseif (ICONV_IMPL === 'glibc') {
                 $s = iconv('UTF-8', 'ASCII//TRANSLIT', $s);
             } else {
-                $s = preg_replace_callback('/[^\x00-\x7F]/u', static function ($c) {
-                    $c = iconv('UTF-8', 'ASCII//IGNORE//TRANSLIT', $c[0]);
+                $s = @preg_replace_callback('/[^\x00-\x7F]/u', static function ($c) {
```

Comment:
why silencing the whole `preg_replace_callback` call ? shouldn't we silence the `iconv` call only ?

Reply:
This makes toggling the error reporting mode only once while doing it in the callback toggles one per match.